### PR TITLE
ChannelContext: fix sendwindow implementation

### PR DESCRIPTION
**Pull request by tmds**

### CODE REVIEW

File: src/Tmds.Ssh/SshClient.ChannelContext.cs [hunk 1]
```diff
@@ -365,25 +353,16 @@ public override async ValueTask SendChannelDataAsync(ReadOnlyMemory<byte> memory
...
-                    if (sendWindow > 0)
+                    try
                     {
-                        int toSend = Math.Min(sendWindow, memory.Length);
-                        toSend = Math.Min(toSend, RemoteMaxPacketSize);
-                        if (Interlocked.CompareExchange(ref _sendWindow, sendWindow - toSend, sendWindow) == sendWindow)
+                        int toSend = await _sendWindow.AquireAsync(aquireCount: memory.Length, exactCount: false, ChannelStopped, ct).ConfigureAwait(false);
+                        await this.SendChannelDataMessageAsync(memory.Slice(0, toSend), ct).ConfigureAwait(false);
+                        memory = memory.Slice(toSend);
+                        if (memory.IsEmpty)
```
Comment (HoundThe) [inline src/Tmds.Ssh/SshClient.ChannelContext.cs:361]: Isn't the condition unnecessary?

File: src/Tmds.Ssh/MultiSemaphore.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
+// This file is part of Tmds.Ssh which is released under MIT.
+// See file LICENSE for full license details.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Tmds.Ssh
+{
+    sealed class MultiSemaphore : IDisposable
```
Comment (HoundThe) [inline src/Tmds.Ssh/MultiSemaphore.cs:10]: Isn't there a Producer/Consumer collection implementation like https://docs.microsoft.com/cs-cz/dotnet/standard/collections/thread-safe/blockingcollection-overview that might do the same thing?

File: src/Tmds.Ssh/SshClient.ChannelContext.cs (hunk 1 above)
Comment (tmds) [reply src/Tmds.Ssh/SshClient.ChannelContext.cs]: :+1: I've removed it.

File: src/Tmds.Ssh/MultiSemaphore.cs (hunk 2 above)
Comment (tmds) [reply src/Tmds.Ssh/MultiSemaphore.cs:11]: Those need real elements. We track space available for send. We don't have something to store.

**Pull request merged**